setup.py

```diff
@@ -1,6 +1,9 @@
 #!/usr/bin/python
 """Upload videos to Youtube."""
-from distutils.core import setup
+try:
+    from setuptools import setup
+except ImportError:
+    from distutils.core import setup
```

Comment:
The `distutils` version of `setup()` [ignores `install_requires`](https://docs.python.org/3/distutils/apiref.html#distutils.core.setup). Python does not yet have a nice way of specifying built-time requirements, as `setuptools` is in this case. The user-friendliest thing you can do is something like this:

```py
import sys
try:
    from setuptools import setup
except ImportError:
    sys.exit("setuptools is required for building youtube-upload")
```